src/algorithms/primaldual.jl

```diff
@@ -186,33 +225,44 @@ struct AFBA{R}
 end
 
 function (solver::AFBA{R})(x0::AbstractArray{C}, y0::AbstractArray{C};
-    f=Zero(), g=Zero(), h=Zero(), l=IndZero(), L=I, betaQ::R=R(0), betaR::R=R(0)
+    f=Zero(), g=Zero(), h=Zero(), l=IndZero(), L=I, beta_f::R=R(0), beta_l::R=R(0)
 ) where {R, C <: Union{R, Complex{R}}}
 
     stop(state::AFBA_state) = norm(state.FPR_x, Inf) + norm(state.FPR_y, Inf) <= solver.tol
     disp((it, state)) = @printf(
         "%6d | %7.4e\n",
-        it, norm(state.FPR_x, Inf)+norm(state.FPR_y, Inf)
+        it, norm(state.FPR_x, Inf) + norm(state.FPR_y, Inf)
     )
 
+    if isa(h, Zero) # case h(Lx) equiv 0
+        L = R(0)*I
+        y0 = zero(x0)
```

Comment:
> Indeed it won't affect the primal variable. When relaxation parameter lambda is not equal to 1, then we end up with updating u^+ = (1-lambda) u. So that is why I set it to zero. Otherwise we can just let it scale the duals throughout the iterations.

So this whole check does not really fix a bug, does it @pylat? Just trying to understand. It really just affects the dual variables, which one doesn’t care much about whenever h==0

Reply:
Without L = R(0)*I, it was a bug (it was using the default L= identity). With the default L, we had barx = prox_g( x- \gamma u - \gamma \nabla f(x)) where the dual variable was converging to zero. So it is incorrect without L = 0 but just happened to converge often.

Reply:
Got it! It’s just that h(1*x) = h(0*x) for all x whenever h == 0, so the problems are equivalent in that case (but apparently the iterations are not)